lib/HiD/Role/IsConverted.pm

```diff
@@ -67,16 +67,15 @@ has converted_content => (
   default => sub {
     my $self = shift;
 
-    my $content = $self->content;
-
     # process template directives in posts
     if( $self->isa('HiD::Post' ) and $self->hid->has_processor() ) {
+      my $content = $self->content;
       $self->hid->processor->process(
         \$self->content , $self->template_data_without_content , \$content
       );
     }
 
-    return _convert_by_extension( $content , $self->ext );
+    return $self->convert_by_extension;
```

Comment:
ah, okay, here's a problem (and now I'm glad I looked at why line 73-74 was doing what it's doing. 

`convert_by_extension` still needs to get passed `$content` (and it still needs to be scoped to outside that `if` block) because `processor->process` is dereffing those scalar refs to modify their contents. That is to say, after line 73, `$content` is no longer a copy of `$self->content` anymore, but an updated copy that has been processed by the processor call.

Reply:
Gotcha. Update in 2 minutes.